napari/components/_tests/test_layer_slicer.py

```diff
@@ -244,3 +245,68 @@ def test_slice_layers_async_task_to_layers_lock(layer_slicer):
 
     assert task.result()[layer].id == 1
     assert task not in layer_slicer._layers_to_task
+
+
+def test_slice_layers_exception_main_thread(layer_slicer):
```

Comment:
These two tests didn't end up being particularly interesting, but I think they're worth keeping around to clarify expectations about exceptions.